doc/user/walkthrough/etcd-operator.md

```diff
@@ -0,0 +1,326 @@
+
+### Create and destroy an etcd cluster
+
+```bash
+$ kubectl create -f example/example-etcd-cluster.yaml
+```
+
+A 3 member etcd cluster will be created.
+
+```bash
+$ kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+example-etcd-cluster-0000       1/1       Running   0          1m
+example-etcd-cluster-0001       1/1       Running   0          1m
+example-etcd-cluster-0002       1/1       Running   0          1m
+```
+
+See [client service](../client_service.md) for how to access etcd clusters created by operator.
+
+If you are working with [minikube locally](https://github.com/kubernetes/minikube#minikube) create a nodePort service and test out that etcd is responding:
+
+```bash
+$ kubectl create -f example/example-etcd-cluster-nodeport-service.json
+$ export ETCDCTL_API=3
+$ export ETCDCTL_ENDPOINTS=$(minikube service example-etcd-cluster-client-service --url)
+$ etcdctl put foo bar
+```
+
+Destroy etcd cluster:
+
+```bash
+$ kubectl delete -f example/example-etcd-cluster.yaml
+```
+
+### Resize an etcd cluster
+
+Create an etcd cluster:
+
+```
+$ kubectl apply -f example/example-etcd-cluster.yaml
+```
+
+In `example/example-etcd-cluster.yaml` the initial cluster size is 3.
+Modify the file and change `size` from 3 to 5.
+
+```
+$ cat example/example-etcd-cluster.yaml
+apiVersion: "etcd.database.coreos.com/v1beta2"
+kind: "EtcdCluster"
+metadata:
+  name: "example-etcd-cluster"
+spec:
+  size: 5
+  version: "3.1.8"
+```
+
+Apply the size change to the cluster CR:
+```
+$ kubectl apply -f example/example-etcd-cluster.yaml
+```
+The etcd cluster will scale to 5 members (5 pods):
+```
+$ kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+example-etcd-cluster-0000       1/1       Running   0          1m
+example-etcd-cluster-0001       1/1       Running   0          1m
+example-etcd-cluster-0002       1/1       Running   0          1m
+example-etcd-cluster-0003       1/1       Running   0          1m
+example-etcd-cluster-0004       1/1       Running   0          1m
+```
+
+Similarly we can decrease the size of cluster from 5 back to 3 by changing the size field again and reapplying the change.
+
+```
+$ cat example/example-etcd-cluster.yaml
+apiVersion: "etcd.database.coreos.com/v1beta2"
+kind: "EtcdCluster"
+metadata:
+  name: "example-etcd-cluster"
+spec:
+  size: 3
+  version: "3.1.8"
+```
+```
+$ kubectl apply -f example/example-etcd-cluster.yaml
+```
+
+We should see that etcd cluster will eventually reduce to 3 pods:
+
+```
+$ kubectl get pods
+NAME                            READY     STATUS    RESTARTS   AGE
+example-etcd-cluster-0002       1/1       Running   0          1m
+example-etcd-cluster-0003       1/1       Running   0          1m
+example-etcd-cluster-0004       1/1       Running   0          1m
+```
+
+### Member recovery
+
+If the minority of etcd members crash, the etcd operator will automatically recover the failure.
```

Comment:
`the etcd operator will  automatically recover` -> `the etcd operator automatically recovers`